packages/integrations/sitemap/src/index.ts

```diff
@@ -25,11 +47,15 @@ export default function createPlugin(): AstroIntegration {
 				config = _config;
 			},
 			'astro:build:done': async ({ pages, dir }) => {
-				const finalSiteUrl = config.site;
+				const finalSiteUrl = canonicalURL || config.site;
 				if (!finalSiteUrl) {
+					console.warn('The Sitemap integration requires either the `site` astro.config option or `canonicalURL` integration option. Skipping.');
```

Comment:
We don't have access to the `logger` from integrations. Happy to remove this if we don't want `console`!

Reply:
This is definitely something to tackle as we improve these APIs. I think `console` is fine for now!

Reply:
Yeah, maybe we could expose Astro's `logger` from the `config:setup` hook?